### PR TITLE
Bugfix/merge on chunks fix

### PR DESCRIPTION
**Pull request by THOR300**
### This Pull Request: 
---
Integrates a bug fix for the error where on large documents we were not merging large documents post chunking correctly. 

I.e. the pages were not being merged.

### CODE REVIEW

File: src/azure_pdf_parser/utils.py [hunk 1]
```diff
@@ -71,6 +71,11 @@ def propagate_page_number(batch: PDFPagesBatchExtracted) -> PDFPagesBatchExtract
                     bounding_region.page_number = (
                         bounding_region.page_number + page_offset
                     )
+
+    if batch.extracted_content.pages:
```
Comment (joel-wright) [inline src/azure_pdf_parser/utils.py:75]: nit: this `if` isn't required, but we can fix that in the follow-up to add a regression test

File: src/azure_pdf_parser/utils.py [hunk 2]
```diff
@@ -92,16 +97,20 @@ def merge_responses(batches: Sequence[PDFPagesBatchExtracted]) -> AnalyzeResult:
 
     all_paragraphs = []
     all_tables = []
+    all_pages = []
     for batch in batches:
         if batch.extracted_content.paragraphs:
             all_paragraphs.extend(batch.extracted_content.paragraphs)
         if batch.extracted_content.tables:
             all_tables.extend(batch.extracted_content.tables)
+        if batch.extracted_content.pages:
```
Comment (joel-wright) [inline src/azure_pdf_parser/utils.py:106]: again, `if` not required, this is guaranteed to be a list

**Review by joel-wright — APPROVED** [review]
This looks like it fixes the immediate problem. We can follow up with code tweaks & regression tests when this is running again.

**Review by diversemix — APPROVED** [review]
lgtm

**Comment by diversemix** [comment]
I would like to see a test that explicitly unit tests the page split / merge functionality :+1:

**Pull request merged**